web/src/user_topic_popover.ts

```diff
@@ -28,14 +43,18 @@ export function initialize(): void {
         onShow(instance) {
             popover_menus.popover_instances.change_visibility_policy = instance;
             popover_menus.on_show_prep(instance);
-            const $elt = $(instance.reference).closest(".change_visibility_policy").expectOne();
-            const stream_id_str = $elt.attr("data-stream-id");
-            $elt.addClass("visibility-policy-popover-visible");
-            assert(stream_id_str !== undefined);
+            const $reference = $(instance.reference);
+            const $change_visibility_policy_button = $reference
+                .closest(".change_visibility_policy")
+                .expectOne();
 
-            const stream_id = Number.parseInt(stream_id_str, 10);
-            const topic_name = $elt.attr("data-topic-name")!;
+            const $data_element = $reference.hasClass("recipient-bar-control")
+                ? $reference.closest(".recipient_bar_controls").expectOne()
+                : $change_visibility_policy_button;
```

Comment:
I don't understand this `?`  comprehension -- I don't see how there's two cases here. If this code is correct, which I'm not sure it is, it definitely deserves a comment.

Reply:
I generally prefer proper if/else blocks for logic that's getting convoluted. But I may be able to live with this one.